corehq/apps/domain/templates/login_and_password/login.html

```diff
@@ -36,3 +36,17 @@
     {% include "login_and_password/partials/login_full.html" %}
   {% endblock %}
 {% endblock content %}
+{% block js-inline %}{{ block.super }}
+  <script>
+    document.addEventListener("DOMContentLoaded", function () {
+      const urlParams = new URLSearchParams(window.location.search);
+      const username = urlParams.get('username');
+      if (username) {
+        var usernameElt = document.getElementById('id_auth-username');
+        if (usernameElt) {
+          usernameElt.value = username;
+        }
+      }
+    });
+  </script>
```

Comment:
Couldn't this go in the JS file for this page?

Reply:
yeah this is a better plan, thanks

Reply:
https://github.com/dimagi/commcare-hq/pull/27000/commits/d2a1cb99b7d27a7b2bbda21fb8d88584cce421aa